plugins/auth-react/src/hooks/useCookieAuthRefresh/useCookieAuthRefresh.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState, useCallback } from 'react';
+import { ApiRef, useApi } from '@backstage/core-plugin-api';
+import { useAsync, useMountEffect } from '@react-hookz/web';
+import { AuthApi } from '../../types';
+
+type CookieAuthRefreshMessage = MessageEvent<{
+  action: string;
+  payload: {
+    expiresAt: string;
+  };
+}>;
+
+/**
+ * @public
+ * It receives an `apiRef` as options, and expects that apiRef extends the `AuthApi` interface.
+ * @remarks
+ * This hook expects a `BroadcastChannel` to be available in the global scope.
+ */
+export function useCookieAuthRefresh<T extends AuthApi>({
+  apiRef,
```

Comment:
Honestly I'm thinking we can take this all the way to something like

```ts
function useCookieAuthRefresh({ pluginId, path = '/cookie' }: { pluginId: string, path?: string }): ...
```

or something like that? Requiring an API might actually be more of a barrier than help in most cases. 

Given that this is a hook we could make it a callback too, which is a bit more flexible, with that I'm thinking something like

```ts
function useCookieAuthRefresh(options: { fetchCookie: () => Promise<{ expiresAt: Date }>}): ...
```

That doesn't translate as well to the provider version though since you're then kinda forced to wrap the provider in your own component to get access to APIs etc.

Reply:
I was going to comment the same. `pluginId: string, options?: { path?: string; }` is a good start, using plain `fetchApi` internally. And then if we ever get the need to make that even more extensible, then we can add a callback that accepts an `ApiHolder` or something too. But let's not worry about that for now I'd say.

Reply:
Great idea, let me try that 🙌🏻

Reply:
Done, it is read for re-review 🙂